service/host.go

```diff
@@ -181,17 +181,29 @@ func (uis *UIServer) modifyHost(w http.ResponseWriter, r *http.Request) {
 		PushFlash(uis.CookieStore, r, w, NewSuccessFlash(msg))
 		gimlet.WriteJSON(w, api.HostStatusWriteConfirm)
 	case "restartJasper":
-		if err = h.SetNeedsJasperRestart(u.Username()); err != nil {
-			gimlet.WriteResponse(w, gimlet.MakeTextInternalErrorResponder(err))
+		var statusCode int
+		statusCode, err = api.GetRestartJasperCallback(r.Context(), env, u.Username())(h)
+		if err != nil {
+			gimlet.WriteResponse(w, gimlet.MakeTextErrorResponder(gimlet.ErrorResponse{
+				StatusCode: statusCode,
+				Message:    err.Error(),
+			}))
 			return
 		}
+
 		PushFlash(uis.CookieStore, r, w, NewSuccessFlash(api.HostRestartJasperConfirm))
 		gimlet.WriteJSON(w, api.HostRestartJasperConfirm)
 	case "reprovisionToNew":
-		if err := h.SetNeedsReprovisionToNew(u.Username()); err != nil {
-			gimlet.WriteResponse(w, gimlet.MakeTextInternalErrorResponder(err))
+		var statusCode int
+		statusCode, err = api.GetReprovisionToNewCallback(r.Context(), env, u.Username())(h)
```

Comment:
won't this be a waste of time if the host is an evergreen host and the agent monitor is running?

Reply:
The first step [to set the DB state](https://github.com/evergreen-ci/evergreen/pull/4418/files#diff-32f1f50563beb33e07a27624dc23580b933aa982fbb7ae04ac3660378d494621R326) still has to occur.

However, it is true that the job can only run if the agent monitor is stopped so to avoid unnecessary work, I added a check to the callback to ensure that the agent monitor is stopped on agent hosts.